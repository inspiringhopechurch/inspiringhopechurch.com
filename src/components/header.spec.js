import React from "react";
import { shallow, mount } from "enzyme";
import Header from "./header";

describe("Header", () => {
  const handleMenu = jest.fn();
  let menuActive;

  beforeEach(() => {
    menuActive = false;
  });

  it("should render without crashing", () => {
    const myHeader = shallow(<Header mobileMenuHandler={handleMenu} mobileMenuActive={menuActive} />);
    expect(myHeader).toMatchSnapshot();
  });

  it("should change mobile menu icon and show menu on click", () => {
    const myHeader = mount(<Header mobileMenuHandler={handleMenu} mobileMenuActive={menuActive} />);

    expect(myHeader.find("button.is-active")).toHaveLength(0);
    expect(myHeader.find("#nav-menu.is-active")).toHaveLength(0);
    expect(handleMenu).toHaveBeenCalledTimes(0);
    myHeader.find("button").simulate("click");
    expect(handleMenu).toHaveBeenCalledTimes(1);
    expect(myHeader.prop("mobileMenuActive")).toBe(menuActive);
    myHeader.setProps({ mobileMenuActive: !menuActive });
    expect(myHeader.prop("mobileMenuActive")).toBe(!menuActive);
    expect(myHeader.find("button.is-active")).toHaveLength(1); // icon change
    expect(myHeader.find("#nav-menu.is-active")).toHaveLength(1); // show hidden menu
  });
});
