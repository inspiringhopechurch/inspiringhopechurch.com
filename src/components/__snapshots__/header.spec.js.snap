// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render without crashing 1`] = `
<header>
  <nav
    aria-label="main navigation"
    className="navbar is-fixed-top"
    role="navigation"
  >
    <div
      className="container px-2"
    >
      <div
        className="navbar-brand"
      >
        <mockConstructor
          className="navbar-item"
          to="/"
        >
          <img
            alt="Inspiring Hope Logo"
            src="test-file-stub"
          />
        </mockConstructor>
        <button
          aria-expanded="false"
          aria-label="menu"
          className="navbar-burger "
          data-target="nav-menu"
          onClick={[MockFunction]}
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </button>
      </div>
      <div
        className="navbar-menu "
        id="nav-menu"
      >
        <div
          className="navbar-start pr-5"
        >
          <mockConstructor
            className="navbar-item"
            to="/"
          >
            Home
          </mockConstructor>
          <div
            className="navbar-item has-dropdown is-hoverable"
          >
            <mockConstructor
              className="navbar-link"
              to="/about"
            >
              About Us
            </mockConstructor>
            <div
              className="navbar-dropdown"
            >
              <mockConstructor
                className="navbar-item"
                to="/about"
              >
                Who We Are
              </mockConstructor>
              <mockConstructor
                className="navbar-item"
                to="/about/beliefs"
              >
                Our Beliefs
              </mockConstructor>
              <mockConstructor
                className="navbar-item"
                to="/about/mission"
              >
                Our Mission
              </mockConstructor>
            </div>
          </div>
          <a
            className="navbar-item"
            href="https://cms.inspiringhopechurch.com/"
          >
            News
          </a>
          <mockConstructor
            className="navbar-item"
            to="/contact"
          >
            Contact
          </mockConstructor>
        </div>
      </div>
    </div>
  </nav>
</header>
`;
